app/controllers/checkout_controller.rb

```diff
@@ -58,6 +61,23 @@ def update
 
   private
 
+  def set_default_bill_address
+    if params[:order][:default_bill_address] == 'YES'
+      new_bill_address = @order.bill_address.clone
+      spree_current_user.set_bill_address(new_bill_address)
+      @order.customer.update_attribute(:bill_address, new_bill_address) unless @order.customer.bill_address
+    end
+
+  end
+
+  def set_default_ship_address
+    if params[:order][:default_ship_address] == 'YES'
+      new_ship_address = @order.ship_address.clone
+      spree_current_user.set_ship_address(new_ship_address)
+      @order.customer.update_attribute(:ship_address, new_ship_address) unless @order.customer.ship_address
```

Comment:
Hm, I don't think that is quite right. If the customer's address was set from admin, we still want the customer to be able to change it if they want to. 

If no customer ship address is currently set and default_ship_address is 'YES' here then we can just update the user's address, because next time the logic in `app/controllers/checkout_controller.rb` will load the user's ship address.
However, if the customer ship address is set (address1), then we need to update address1 to whatever the order address is (address2) here when `default_ship_address == 'YES'`. Otherwise, the next time they go through the checkout, the logic will load address1 and not address2 as they would expect.

What do you think?